python/ray/data/read_api.py

```diff
@@ -2313,7 +2317,13 @@ def from_torch(
     Returns:
         A :class:`MaterializedDataset` containing the Torch dataset samples.
     """  # noqa: E501
-    return from_items(list(dataset))
+    return read_datasource(
```

Comment:
we should only support parallel reads for map-style PyTorch Datasets. For iterable-style pytorch datasets, reading in parallel will read the entire Dataset N times. For this case, we should only read in 1 read task.